mlshvch/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+file = File.open('data.txt', 'r')
+data = file.read
+file.close
```

Comment:
Relative paths does not play good when you are trying to start your script not from the exact directory where it is located. For example, it does not work, when I try to run it from `mlshvch/` folder:
```bash
> pwd
Ruby-Lab/mlshvch
> hometask-3/rainfall.rb
Traceback (most recent call last):
	2: from hometask-3/rainfall.rb:5:in `<main>'
	1: from hometask-3/rainfall.rb:5:in `open'
hometask-3/rainfall.rb:5:in `initialize': No such file or directory @ rb_sysopen - data.txt (Errno::ENOENT)
```

A possible solution there is to use the possibilities of `File` Ruby class and `__dir__` / `__FILE__` constants to build up a correct absolute path to a file.

Reply:
Also, the block notation of `File.open` closes file automatically at the end of the block. If you use that you save your time and don't worry about file handlers. As another side note: `File#read` does everything for you: opens a file, reads the contents and closes it afterwards. Take a look ;)

Reply:
Also, this could be wrapped in a function, there is no need for it to be called in the void.